pkg/virt-handler/vm.go

```diff
@@ -2385,6 +2394,22 @@ func (d *VirtualMachineController) vmUpdateHelperMigrationSource(origVMI *v1.Vir
 			d.recorder.Event(vmi, k8sv1.EventTypeNormal, v1.Migrating.String(), "VirtualMachineInstance is aborting migration.")
 		}
 	} else {
+		if isMigrationInProgress(vmi, domain) {
+			// we already started this migration, no need to rerun this
+			log.DefaultLogger().Errorf("migration %s has already been started", vmi.Status.MigrationState.MigrationUID)
+			return nil
+		}
+
+		err = hostdisk.ReplacePVCByHostDisk(vmi)
```

Comment:
Nit: I would move this closer to the `MigrateVirtualMachine` call.

Reply:
Why?

Reply:
Afaik, It is only needed for that call, and having it closer could make it more oblivious.